webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx

```diff
@@ -147,8 +161,16 @@ const ModalForm: React.FC<Props> = ({ form, user, groups, viewOnly }) => {
         </Form.Item>
       )}
       {rbacEnabled && canModifyPermissions && !viewOnly && (
-        <Form.Item label={ROLE_LABEL} name={ROLE_NAME}>
-          <Select mode="multiple" optionFilterProp="children" placeholder={'Add Roles'} showSearch>
+        <Form.Item
+          initialValue={roles === null ? [] : roles.map((r) => r.id)}
+          label={ROLE_LABEL}
+          name={ROLE_NAME}>
+          <Select
+            disabled={user !== undefined && roles === null}
+            mode="multiple"
+            optionFilterProp="children"
+            placeholder={'Add Global Roles'}
+            showSearch>
             {knownRoles.map((r) => (
```

Comment:
Can we refer to the filter [here](https://github.com/determined-ai/determined/blob/85396ebc4f0ec95c6b4ef9625061c92fa324cdce/webui/react/src/hooks/useModal/UserSettings/useModalGroupRoles.tsx#L68) to only show the global roles

Reply:
Done.